src/slidepane/tests/functional/SlidePane.ts

```diff
@@ -0,0 +1,192 @@
+			.moveMouseTo(null, 300, 10)
+			.clickMouseButton()
+			.sleep(DELAY)
+			.end();
+	}
+
+	return underlay
+		.findByCssSelector(UNDERLAY_SELECTOR)
+			.click()
+			.sleep(DELAY)
+			.end();
+}
+
+function swipeSlidePane(remote: any, distance = 250, alignRight?: boolean) {
+	const { mouseEnabled } = remote.environmentType;
+	const initialX = alignRight ? 10 : 300;
+
+	if (mouseEnabled) {
+		const finalX = alignRight ? distance : -distance;
+		return openSlidePane(remote, alignRight)
+			.moveMouseTo(null, initialX, 10)
+			.pressMouseButton()
+			.moveMouseTo(null, finalX, 0)
+			.releaseMouseButton()
+			.sleep(DELAY);
+	}
+
+	const finalX = alignRight ? initialX + distance : initialX - distance;
+	return openSlidePane(remote, alignRight)
+		.pressFinger(initialX, 10)
+		.moveFinger(finalX, 10)
+		.releaseFinger(finalX, 10)
+		.sleep(DELAY);
+}
+
+registerSuite({
+	name: 'SlidePane',
+
+	'the underlay should cover the screen'(this: any) {
+		let viewportSize: { height: number; width: number; };
+
+		return openSlidePane(this.remote)
+			.getWindowSize()
+				.then(({ height, width }: { height: number; width: number; }) => {
+					viewportSize = { height, width };
+				})
+			.findByCssSelector(UNDERLAY_SELECTOR)
+				.getSize()
+				.then(({ height, width }: { height: number; width: number; }) => {
+					assert.closeTo(height, viewportSize.height, viewportSize.height * 0.2);
+					assert.closeTo(width, viewportSize.width, viewportSize.width * 0.2);
+				});
+	},
+
+	'the underlay should be destroyed when the slidepane is hidden'(this: any) {
+		return clickUnderlay(this.remote)
+			.findByCssSelector(UNDERLAY_SELECTOR)
+				.getAttribute('class')
+				.then((className: string) => {
+					assert.match(className, /slidePane-m__content/, 'the underlay should be removed.');
+				});
+	},
+
+	'the underlay should not be destroyed when the slidepane is clicked'(this: any) {
+		return openSlidePane(this.remote)
+			.findByCssSelector(SLIDEPANE_SELECTOR)
+				.click()
+				.end()
+			.sleep(DELAY)
+			.findByCssSelector(UNDERLAY_SELECTOR)
+				.getAttribute('class')
+				.then((className: string) => {
+					assert.match(className, /slidePane-m__underlay/, 'the underlay should not be removed.');
+				});
+	},
+
+	'the slidepane should not be hidden when it is clicked'(this: any) {
+		return openSlidePane(this.remote)
+			.findByCssSelector(SLIDEPANE_SELECTOR)
+				.click()
+				.sleep(DELAY)
+				.getPosition()
+					.then(({ x }: { x: number; }) => {
+						assert.strictEqual(x, 0, 'The slidepane should be visible.');
+					});
+	},
+
+	'a left-aligned slidepane should close when swiping from right to left'(this: any) {
+		const { browserName, mouseEnabled, touchEnabled } = this.remote.environmentType;
+
+		if (!mouseEnabled && !touchEnabled) {
+			this.skip('Test requires mouse or touch interactions.');
+		}
+
+		if (browserName.toLowerCase() === 'microsoftedge') {
+			this.skip('Edge driver does not handle mouse movements correctly.');
```

Comment:
Is there an open issue with Edge that we can link to with a TODO to track for the future?